src/jaxsim/mujoco/loaders.py

```diff
@@ -0,0 +1,480 @@
+            gazebo_preserve_fixed_joints=False,
+            pretty=True,
+        )
+
+        # -------------------------------------
+        # Add a floating joint if floating-base
+        # -------------------------------------
+
+        if not rod_model.is_fixed_base():
+            considered_joints |= {"world_to_base"}
+            urdf_string = RodModelToMjcf.add_floating_joint(
+                urdf_string=urdf_string,
+                base_link_name=rod_model.get_canonical_link(),
+                floating_joint_name="world_to_base",
+            )
+
+        # ---------------------------------------
+        # Inject the <mujoco> element in the URDF
+        # ---------------------------------------
+
+        parser = ET.XMLParser(remove_blank_text=True)
+        root = ET.fromstring(text=urdf_string.encode(), parser=parser)
+
+        mujoco_element = (
+            ET.SubElement(root, "mujoco")
+            if len(root.findall("./mujoco")) == 0
+            else root.find("./mujoco")
+        )
+
+        _ = ET.SubElement(
+            mujoco_element,
+            "compiler",
+            balanceinertia="true",
+            discardvisual="false",
+        )
+
+        urdf_string = ET.tostring(root, pretty_print=True).decode()
+        # print(urdf_string)
+        # raise
+
+        # ------------------------------
+        # Post-process all dummy visuals
+        # ------------------------------
+
+        parser = ET.XMLParser(remove_blank_text=True)
+        root: ET._Element = ET.fromstring(text=urdf_string.encode(), parser=parser)
+        import numpy as np
+
+        # Give a tiny radius to all dummy spheres
+        for geometry in root.findall(".//visual/geometry[sphere]"):
+            radius = np.fromstring(
+                geometry.find("./sphere").attrib["radius"], sep=" ", dtype=float
+            )
+            if np.allclose(radius, np.zeros(1)):
+                geometry.find("./sphere").set("radius", "0.001")
+
+        # Give a tiny volume to all dummy boxes
+        for geometry in root.findall(".//visual/geometry[box]"):
+            size = np.fromstring(
+                geometry.find("./box").attrib["size"], sep=" ", dtype=float
+            )
+            if np.allclose(size, np.zeros(3)):
+                geometry.find("./box").set("size", "0.001 0.001 0.001")
+
+        urdf_string = ET.tostring(root, pretty_print=True).decode()
+
+        # ------------------------
+        # Convert the URDF to MJCF
+        # ------------------------
+
+        # Load the URDF model into Mujoco.
+        assets = RodModelToMjcf.assets_from_rod_model(rod_model=rod_model)
+        mj_model = mj.MjModel.from_xml_string(xml=urdf_string, assets=assets)  # noqa
+
+        # Get the joint names.
+        mj_joint_names = set(
+            [
+                mj.mj_id2name(mj_model, mj.mjtObj.mjOBJ_JOINT, idx)
+                for idx in range(mj_model.njnt)
+            ]
+        )
```

Comment:
```suggestion
        mj_joint_names = {
            mj.mj_id2name(mj_model, mj.mjtObj.mjOBJ_JOINT, idx)
            for idx in range(mj_model.njnt)
        }
```
You could directly use a set comprehension here

Reply:
Addressed in https://github.com/ami-iit/jaxsim/pull/83/commits/4dcd4a760ac2d9afe7d6a74fea385f2447ebb07f